pyproject.toml

```diff
@@ -26,7 +26,7 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
 ]
 dynamic = ["version"]
-requires-python = ">=3.8"
+requires-python = ">=3.9"
```

Comment:
Given the failure on python v3.12, should we change this to `>=3.9, <3.12`?

I think we should go forward as-is without >3.11 support, but given that Python currently [supports versions](https://devguide.python.org/versions/) up to 3.13, I would like to not fall that far behind. Maybe I can add a PR on [lspopt](https://github.com/hbldh/lspopt) about this? We could be patient about a fix and then come up with an alternate plan if need be.

Reply:
I would keep support for 3.12. I'll investigate the lspopt issue now

Reply:
Problem solved here as well. Just needed to add `setuptools` to the dependencies.

Reply:
Note that this still creates a DeprecationWarning in the tests. We should update lspopt at some point:

```
=============================== warnings summary ===============================
../../../../../opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/lspopt/data/__init__.py:11
  /opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/lspopt/data/__init__.py:11: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import resource_filename
```